### ISSUE TITLE
Cached folders are not included in a task's output

### ISSUE THREAD

**Issue by mattdodge** [concourse/concourse#5933]
## Summary

If a task has a folder cached in the same folder that is output the subsequent tasks do not receive the output correctly. It's probably rare to have cached folders in an *output* but it seems strange that a cached folder would be removed from them.

## Steps to reproduce

My use case is around building a node app and testing it. I know there are different ways to do this but here is what I was trying to do where I noticed the strange behavior.

```yml
plan:
  - task: build
    config:
      caches:
        - path: app/node_modules
      inputs:
        - name: app
      outputs:
        - name: app_built
          path: app
      run: yarn install && yarn build   # pseudo code
  - task: test
    config:
      inputs:
        - name: app_built
          path: app
      run: yarn test
```

Basically my app code comes in as an input. I want to install node dependencies (with a cache) and then output the built folder. Outputting an input this way may be slightly undefined behavior 😕 

## Expected/Actual results

I would expect the input of my second task to have a populated `app/node_modules` folder. Instead, the folder is empty. If I do not cache the `app/node_modules` folder in the first task it works as expected.

## Triaging info

* Concourse version: 6.1.0
* Did this used to work?: Unsure, not that I know of

**Comment by taylorsilva** [comment]
Agreed, weird behaviour from a user's POV. Probably an issue with how we layer the cache volume onto the underlying input volume. I'd guess they're two different mount points.

```
app/ <-- one mount point
   node_modules/ <-- second mount point
```

In this example, only the first mount point gets passed as the output.

Since the workaround is to not flag the folder as cached I'm closing this.
